cmd/loom/gateway/mapping_cmds.go

```diff
@@ -265,6 +266,136 @@ func newMapAccountsCommand() *cobra.Command {
 	return cmd
 }
 
+type ConfirmedMapping struct {
+	From string `json:"from"`
+	To   string `json:"to"`
+}
+
+type PendingMapping struct {
+	ForeignContract string `json:"foreign_contract"`
+	LocalContract   string `json:"local_contract"`
+}
+
+type Mappings struct {
+	ConfirmedMappings []ConfirmedMapping `json:"confirmed_mapping"`
+	PendingMappings   []PendingMapping   `json:"pending_mapping"`
+}
+
+const ListContractMappingCmdExample = `
+loom gateway list-contract-mappings
+`
+
+func newListContractMappingsCommand() *cobra.Command {
+	var gatewayType string
+	cmd := &cobra.Command{
+		Use:     "list-contract-mappings",
+		Short:   "List all contract mappings",
+		Example: ListContractMappingCmdExample,
+		Args:    cobra.MinimumNArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var mapping Mappings
+			rpcClient := getDAppChainClient()
+			gatewayAddr, err := rpcClient.Resolve(gatewayType)
+			if err != nil {
+				return errors.Wrap(err, "failed to resolve DAppChain Gateway address")
+			}
+			gateway := client.NewContract(rpcClient, gatewayAddr.Local)
+			req := &tgtypes.TransferGatewayListContractMappingRequest{}
+			resp := &tgtypes.TransferGatewayListContractMappingResponse{}
+			_, err = gateway.StaticCall("ListContractMapping", req, gatewayAddr, resp)
+			if err != nil {
+				return errors.Wrap(err, "failed to call gateway.ListContractMapping")
+			}
+			confirmedmappings := []ConfirmedMapping{}
+			for _, mapping := range resp.ConfimedMappings {
+				confirmedmappings = append(confirmedmappings, getConfirmedMappingInfo(mapping))
+			}
+			pendingmappings := []PendingMapping{}
+			for _, mapping := range resp.PendingMappings {
+				pendingmappings = append(pendingmappings, getPendingMappingInfo(mapping))
+			}
+			mapping.PendingMappings = pendingmappings
+			mapping.ConfirmedMappings = confirmedmappings
+			output, err := json.MarshalIndent(mapping, "", "  ")
+			if err != nil {
+				return err
+			}
+			fmt.Println(string(output))
+			return nil
+		},
+	}
+	cmdFlags := cmd.Flags()
+	cmdFlags.StringVar(&gatewayType, "gateway", "gateway", "Gateway name: gateway, loomcoin-gateway, or tron-gateway")
+	return cmd
+}
+
+const getContractMappingCmdExample = `
+loom gateway get-contract-mapping 0x7262d4c97c7B93937E4810D289b7320e9dA82857
+`
+
+type Mapping struct {
+	Address   string `json:"address"`
+	IsPending bool   `json:"is_pending"`
+	Found     bool   `json:"found"`
+}
+
+func newGetContractMappingCommand() *cobra.Command {
+	var gatewayType string
+	cmd := &cobra.Command{
+		Use:     "get-contract-mapping <contract-addr>",
+		Short:   "Get Contract Mapping",
+		Example: getContractMappingCmdExample,
+		Args:    cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var mapping Mapping
+			var contractAddr loom.Address
+			var err error
+			if strings.HasPrefix(args[0], "eth:") {
```

Comment:
This shouldn't be limited to `eth:`, we've also got `tron:` and `binance:`, and probably others, so lets make sure this works with any chain ID.

Reply:
Done.